docker-compose.yml

```diff
@@ -97,6 +99,7 @@ services:
                     DOCKER_COMPOSE_DIRECTORY: $PWD
                     DEVICE_HOST: ${DEVICE_HOST:-http://umbrel.local}
                     MIDDLEWARE_API_URL: "http://10.11.2.2"
+                    UMBREL_SEED_FILE: "/db/umbrel-seed/seed"
```

Comment:
Sounds interesting.

We've already had some internal discussion on how to transition to alternative entropy sources in the future.

For example this PR is using `umbrel-seed` that is persisted on disk for non-interactive use-cases. e.g automated backups without requiring user interaction.

There could be another root seed, `umbrel-interactive-seed` that requires `umbrel-seed` + user password + `usecase indentifier` for derivation. This would never be stored on disk. So each time an app requires it, the user is prompted for their password to derive the entropy for that specific use case, the entropy is in memory while it's being used, and then destroyed.

Reply:
> For example this PR is using umbrel-seed that is persisted on disk for non-interactive use-cases. e.g automated backups without requiring user interaction.

If the need to encrypt is the only secret needed for backup using asymmetric crypto can solve it altogether (totally not shilling [`age`] here again).

[`age`]: https://github.com/FiloSottile/age


> There could be another root seed, umbrel-interactive-seed that requires umbrel-seed + user password + usecase indentifier for derivation. This would never be stored on disk. So each time an app requires it, the user is prompted for their password to derive the entropy for that specific use case, the entropy is in memory while it's being used, and then destroyed.

Not sure if you caught that contents of `/run/` are in RAM only already 🤔.  So that all decrypted stuff is always ephemeral.  What can also be done is to have said `tmpfs` encrypted, so even if RBP gets deep-frozen extracting secrets from RAM becomes _less easy_.

Reply:
Age was actually what I initially planned to use for this, but due to time constraints just went with symmetric PGP encryption cos it's super simple to use and I'm very familiar with it.

We can completely change the algorithm/keys used for backups at any point in the future without too much trouble btw.